blivet/zfcp.py

```diff
@@ -167,36 +327,9 @@ def online_device(self):
 
         return True
 
-    def offline_scsi_device(self):
-        # A list of existing SCSI devices in format Host:Bus:Target:Lun
-        scsi_devices = [f for f in os.listdir(scsidevsysfs) if re.search(r'^[0-9]+:[0-9]+:[0-9]+:[0-9]+$', f)]
-
-        for scsidev in scsi_devices:
-            fcpsysfs = "%s/%s" % (scsidevsysfs, scsidev)
-            scsidel = "%s/%s/delete" % (scsidevsysfs, scsidev)
-
-            f = open("%s/hba_id" % (fcpsysfs), "r")
-            fcphbasysfs = f.readline().strip()
-            f.close()
-            f = open("%s/wwpn" % (fcpsysfs), "r")
-            fcpwwpnsysfs = f.readline().strip()
-            f.close()
-            f = open("%s/fcp_lun" % (fcpsysfs), "r")
-            fcplunsysfs = f.readline().strip()
-            f.close()
-
-            if fcphbasysfs == self.devnum \
-                    and fcpwwpnsysfs == self.wwpn \
-                    and fcplunsysfs == self.fcplun:
-                logged_write_line_to_file(scsidel, "1")
-                udev.settle()
-                return
-
-        log.warning("no scsi device found to delete for zfcp %s %s %s",
-                    self.devnum, self.wwpn, self.fcplun)
-
     def offline_device(self):
-        offline = "%s/%s/online" % (zfcpsysfs, self.devnum)
+        """Remove the zFCP device from the system."""
```

Comment:
nit pick: Setting an FCP device offline (which was online before) does not remove it from the system, it keeps all the state info and Linux driver core sysfs tree with zfcp_adapter,  zfcp_port-s, and zfcp_unit-s and only sets it into an offline state. (Reason: it can recover the previous state on the next online operation.)
No need to change the comment, though.